File: api/src/main/java/com/delivery/api/business/TokenBusiness.java
package com.delivery.api.business;

import com.delivery.api.converter.TokenConverter;
import com.delivery.api.service.TokenService;
import com.delivery.common.annotation.Business;
import com.delivery.common.dto.TokenResponse;
import com.delivery.common.exception.ApiException;
import com.delivery.common.status.ErrorStatusCode;
import com.delivery.db.entity.User;
import lombok.RequiredArgsConstructor;

import java.util.Optional;

@Business
@RequiredArgsConstructor
public class TokenBusiness {

    private final TokenService tokenService;
    private final TokenConverter tokenConverter;

    public TokenResponse issueToken(User userEntity) {
        return Optional.ofNullable(userEntity)
                .map(item -> {
                    var id = item.getId();
                    var accessToken = tokenService.issueAccessToken(id);
                    var refreshToken = tokenService.issueRefreshToken(id);
                    return tokenConverter.toResponse(accessToken, refreshToken);
                })
                .orElseThrow(() -> new ApiException(ErrorStatusCode.NULL_POINT));
    }

    public Long getIdFromAccessToken(String accessToken) {
        return tokenService.getIdOrThrow(accessToken);
    }
}

File: account/src/main/kotlin/com/delivery/account/business/TokenBusiness.kt
package com.delivery.account.business

import com.delivery.account.service.TokenService
import com.delivery.common.annotation.Business
import com.delivery.common.dto.TokenValidationRequest
import com.delivery.common.dto.TokenValidationResponse

@Business
class TokenBusiness(
    private val tokenService: TokenService
) {
    fun validateToken(request: TokenValidationRequest): TokenValidationResponse {
        val result = tokenService.getIdOrThrow(request.token.token)

        return TokenValidationResponse(
            id = result
        )
    }
}
